### PR TITLE
Fix multiple `copy-file` buttons on gist svg files

### LINKED ISSUE #3323: Multiple copy buttons in SVG file in gist
Three copy buttons appear for an SVG file in a Github Gist. All of them are functional.

[Link][1]
[image][1]

[1]: https://gist.github.com/cimi/8c9674e587d3f05131e42353ab1282c3?short_path=939f372

### PR DESCRIPTION
**Pull request by yakov116**
1. LINKED ISSUES:
   Fixes #3323 

2. TEST URLS:
https://gist.github.com/cimi/8c9674e587d3f05131e42353ab1282c3?short_path=939f372

   https://github.com/jmeridth/jmeridth.github.io/blob/gh-pages/_posts/2012-03-30-do-not-issue-pull-requests-from-your-master-branch.md (No changes)

3. SCREENSHOT:
   Before
image

After

image

### CODE REVIEW

File: source/features/copy-file.tsx [hunk 1]
```diff
@@ -22,18 +23,28 @@ function handleClick({delegateTarget: button}: delegate.Event<MouseEvent, HTMLBu
...
+		const copyButton = (
+			<button
+				className="btn btn-sm tooltipped tooltipped-n BtnGroup-item rgh-copy-file"
+				aria-label="Copy file to clipboard"
+				type="button"
+			>
+				Copy
+			</button>
+		);
+		if (pageDetect.isGist() && button.href.endsWith('.svg')) { // SVG files have a render blob button
```
Comment (yakov116) [inline source/features/copy-file.tsx:39]: Without this

image

File: source/features/copy-file.tsx [hunk 2]
```diff
@@ -22,18 +23,28 @@ function handleClick({delegateTarget: button}: delegate.Event<MouseEvent, HTMLBu
...
+			button.before(copyButton);
+		} else {
+			button
+				.parentElement! // `BtnGroup`
+				.prepend(copyButton);
+		}
+
+		groupSiblings(button);
```
Comment (yakov116) [inline source/features/copy-file.tsx:47]: Without this (effects only gists)

image

File: source/features/copy-file.tsx (hunk 1 above)
Comment (fregante) [reply source/features/copy-file.tsx:39]: > SVG files have a render blob button

It's not exclusive to SVGs; use a DOM-based detection

Comment (yakov116) [reply source/features/copy-file.tsx:39]: Do you have another example that I could test on? I tried a few files and could not find another one.

Comment (fregante) [reply source/features/copy-file.tsx]: Markdown is one, through our `view-markdown-source` feature

Comment (fregante) [reply source/features/copy-file.tsx]: Possibly these too https://docs.github.com/en/github/managing-files-in-a-repository/working-with-non-code-files

Comment (yakov116) [reply source/features/copy-file.tsx]: > Markdown is one, through our `view-markdown-source` feature

`view-markdown-source` is not enabled (and cannot be) on gists.

I did a dom detection, let me know if you want me to improve anything.

File: source/features/copy-file.tsx [hunk 3]
```diff
@@ -22,18 +23,29 @@ function handleClick({delegateTarget: button}: delegate.Event<MouseEvent, HTMLBu
...
+		const hasRenderButtons = button.parentElement!.querySelector('.rendered');
+		if (pageDetect.isGist() && hasRenderButtons) {
+			button.before(copyButton);
+		} else {
+			button
+				.parentElement! // `BtnGroup`
+				.prepend(copyButton);
+		}
+
+		groupSiblings(button);
```
Comment (fregante) [inline source/features/copy-file.tsx:48]: The code expected a group to be there; let's look for it explicitly:

```suggestion
		const group = button.closest('.BtnGroup');
		if (group) {
			group.prepend(copyButton);
		} else {
			groupButtons([button, copyButton]);
		}
```

**Comment by fregante** [comment]
Needs some fixin’

**Comment by yakov116** [comment]
> Needs some fixin’
fixin’ complete

**Pull request merged**